airflow/utils/db.py

```diff
@@ -1683,8 +1686,10 @@ def upgradedb(
                 os.environ["AIRFLOW__DATABASE__SQL_ALCHEMY_MAX_SIZE"] = val
             settings.reconfigure_orm()
 
-    if reserialize_dags:
+    current_revision = to_revision or _get_current_revision(session=session)
```

Comment:
Also - possibly reserializationshould be also protected by the lock?

Reply:
In that case the next two lines probably should as well?

Reply:
Yeah. I'd do that too likely.